### PR TITLE
 Implement a Distribute play store library

### PR DESCRIPTION
**Pull request by AnastasiaKubova**
Please have a look at our [guidelines for contributions](https://github.com/microsoft/appcenter-sdk-android/blob/develop/CONTRIBUTING.md) and consider the following before you submit the PR:

* [ ] ~Has `CHANGELOG.md` been updated?~
* [x] Are tests passing locally?
* [x] Are the files formatted correctly?
* [x] Did you add unit tests?
* [x] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?
* [x] Did you check UI tests on the sample app? They are not executed on CI.

## Description

Implement a Distribute Play store library with stubbed APIs that do nothing.

## Related PRs or issues

[AB#81911](https://msmobilecenter.visualstudio.com/454a20a9-dfe6-4c1f-8ae8-417f76adb472/_workitems/edit/81911)

### CODE REVIEW

File: sdk/appcenter-distribute-play/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+<!--
+  ~ Copyright (c) Microsoft Corporation. All rights reserved.
+  ~ Licensed under the MIT License.
+  -->
+
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
+    package="com.microsoft.appcenter.distribute">
+
+    <uses-permission android:name="android.permission.REQUEST_INSTALL_PACKAGES" />
+    <uses-permission android:name="android.permission.DOWNLOAD_WITHOUT_NOTIFICATION" />
```
Comment (MatkovIvan) [inline sdk/appcenter-distribute-play/src/main/AndroidManifest.xml:11]: This must be removed

File: sdk/appcenter-distribute-play/src/main/java/com/microsoft/appcenter/distribute/Distribute.java [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+    }
+
+    /**
+     * Check whether Distribute service is enabled or not.
+     *
+     * @return future with result being <code>true</code> if enabled, <code>false</code> otherwise.
+     * @see AppCenterFuture
+     */
+    public static AppCenterFuture<Boolean> isEnabled() {
+        AppCenterLog.debug(LOG_TAG, "Called method 'isEnabled'");
```
Comment (MatkovIvan) [inline sdk/appcenter-distribute-play/src/main/java/com/microsoft/appcenter/distribute/Distribute.java:53]: Please clean up this kind of things

File: sdk/appcenter-distribute-play/src/main/java/com/microsoft/appcenter/distribute/Distribute.java [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+    private static Distribute sInstance;
+
+    private static final String DISTRIBUTE_GROUP = "group_distribute";
+
+    /**
+     * Get shared instance.
+     *
+     * @return shared instance.
+     */
+    public static synchronized Distribute getInstance() {
```
Comment (MatkovIvan) [inline sdk/appcenter-distribute-play/src/main/java/com/microsoft/appcenter/distribute/Distribute.java:39]: I remember that in some cases public resume method was used to workaround missing activity if it's started after all events, please add that method too

File: sdk/appcenter-distribute-play/src/main/java/com/microsoft/appcenter/distribute/ReleaseDetails.java [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+    }
+
+    /**
+     * Get the releaseNotes value.
+     *
+     * @return the releaseNotes value
+     */
+    @Nullable
+    public String getReleaseNotes() {
+        return "releaseNotes";
```
Comment (MatkovIvan) [inline sdk/appcenter-distribute-play/src/main/java/com/microsoft/appcenter/distribute/ReleaseDetails.java:63]: null?

File: sdk/appcenter-distribute-play/src/main/java/com/microsoft/appcenter/distribute/ReleaseDetails.java [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+
+    /**
+     * Get the releasesNotesUrl value.
+     *
+     * @return the releaseNotesUrl value.
+     */
+    @Nullable
+    @SuppressWarnings("WeakerAccess")
+    public Uri getReleaseNotesUrl() {
+        return Uri.parse("releaseNotesUrl");
```
Comment (MatkovIvan) [inline sdk/appcenter-distribute-play/src/main/java/com/microsoft/appcenter/distribute/ReleaseDetails.java:74]: null?

File: sdk/appcenter-distribute-play/src/main/java/com/microsoft/appcenter/distribute/ReleaseDetails.java [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+    }
+
+    /**
+     * Get the minApiLevel value.
+     *
+     * @return the minApiLevel value
+     */
+    int getMinApiLevel() {
+        return 16;
```
Comment (MatkovIvan) [inline sdk/appcenter-distribute-play/src/main/java/com/microsoft/appcenter/distribute/ReleaseDetails.java:83]: 0?

File: sdk/appcenter-distribute-play/src/main/java/com/microsoft/appcenter/distribute/ingestion/models/DistributionStartSessionLog.java [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+ */
+
+package com.microsoft.appcenter.distribute.ingestion.models;
+
+import com.microsoft.appcenter.ingestion.models.AbstractLog;
+
+/**
+ * Distribution start session log.
+ */
+public class DistributionStartSessionLog extends AbstractLog {
```
Comment (MatkovIvan) [inline sdk/appcenter-distribute-play/src/main/java/com/microsoft/appcenter/distribute/ingestion/models/DistributionStartSessionLog.java:13]: Why do we need this class?

File: sdk/appcenter-distribute-play/src/main/res/values-cs/appcenter_distribute.xml [hunk 8]
```diff
@@ -0,0 +1,27 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+  ~ Copyright (c) Microsoft Corporation. All rights reserved.
+  ~ Licensed under the MIT License.
+  -->
+
+<resources>
```
Comment (MatkovIvan) [inline sdk/appcenter-distribute-play/src/main/res/values-cs/appcenter_distribute.xml:7]: No need for this string resources

File: settings.gradle [hunk 9]
```diff
@@ -10,6 +10,7 @@ include ':sdk:appcenter-crashes'
 include ':sdk:appcenter-analytics'
 include ':sdk:appcenter-distribute'
 include ':sdk:appcenter-push'
+include ':sdk:appcenter-distribute-play'
```
Comment (MatkovIvan) [inline settings.gradle:13]: Sort

File: sdk/appcenter-distribute-play/src/main/java/com/microsoft/appcenter/distribute/DistributeConstants.java [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+
+/**
+ * Distribute constants.
+ */
+public final class DistributeConstants {
+
+    /**
+     * Distribute service name.
+     */
+    static final String SERVICE_NAME = "DistributePlay";
```
Comment (guperrot) [inline sdk/appcenter-distribute-play/src/main/java/com/microsoft/appcenter/distribute/DistributeConstants.java:20]: Why do we need all the other constants?

File: sdk/appcenter-distribute-play/src/main/res/values/appcenter_distribute.xml [hunk 11]
```diff
@@ -0,0 +1,30 @@
+<?xml version="1.0" encoding="utf-8"?><!--
+  ~ Copyright (c) Microsoft Corporation. All rights reserved.
+  ~ Licensed under the MIT License.
+  -->
+
+<resources xmlns:tools="http://schemas.android.com/tools">
+    <string name="appcenter_distribute_update_dialog_title">App update available</string>
+    <string name="appcenter_distribute_update_dialog_message_optional">%1$s %2$s (%3$d) is available to download and install.</string>
+    <string name="appcenter_distribute_update_dialog_message_mandatory">The developers of this app require you to update to %1$s %2$s (%3$d).</string>
```
Comment (guperrot) [inline sdk/appcenter-distribute-play/src/main/res/values/appcenter_distribute.xml:9]: Should we create a distribute-core to share identical files and avoid duplication?

Comment (MatkovIvan) [reply sdk/appcenter-distribute-play/src/main/res/values/appcenter_distribute.xml:9]: I don't think that we should add one more library and publish one more package to share just a few files. Sharing files like Xamarin would work here, but there is no similar way to do it in native Android

Comment (MatkovIvan) [reply sdk/appcenter-distribute-play/src/main/res/values/appcenter_distribute.xml]: We need to try symlinks for these files

Comment (guperrot) [reply sdk/appcenter-distribute-play/src/main/res/values/appcenter_distribute.xml]: Are symbolic links working on Windows? If not we should not use them and keep this repo compatible with all OS.

Comment (AnastasiaKubova) [reply sdk/appcenter-distribute-play/src/main/res/values/appcenter_distribute.xml:9]: I checked changes on Windows. Looks like the symbolic links for Mac don't work on Windows, I reverted changes.

**Pull request merged**